pkgs/tools/misc/thumbs/default.nix

```diff
@@ -0,0 +1,26 @@
+{ lib, rustPlatform, fetchFromGitHub }:
+rustPlatform.buildRustPackage rec {
+  pname = "thumbs";
+  version = "0.7.1";
+
+  src = fetchFromGitHub {
+    owner = "fcsonline";
+    repo = "tmux-thumbs";
+    rev = version;
+    sha256 = "sha256-PH1nscmVhxJFupS7dlbOb+qEwG/Pa/2P6XFIbR/cfaQ=";
+  };
+
+  cargoSha256 = "sha256-6htKiXMMyYRFefJzvDnmdx3CJ3XL8zONhGlV2wcbr9g=";
+
+  patches = [ ./fix.patch ];
+  postPatch = ''
+    substituteInPlace src/swapper.rs --replace '@@thumbs-bin-dir@@' $out/bin
+  '';
```

Comment:
I am not sure if it is possible, at last I haven't found a way to do this. The substitution uses `out` but I cannot use it as a nix variable because the derivation hasn't been created yet. 

Following code correctly complains about `out` being undefined:
```
 patches = [
    (substituteAll {
      src = ./fix.patch;
      thumbsBinDir = "${out}/bin";
    })
  ];
```

Am I missing something?

Reply:
Unfortunately, neither of them works. My debugging capabilities of that are very limited but I was about to observe following errors:
With first solution:
```
 patches = [
    (substituteAll {
      src = ./fix.patch;
      thumbsBinDir = "$out/bin";
    })
  ];
```
`Not a directory : $out/bin/thumbs`


With second solution:
```
 patches = [
    (substituteAll {
      src = ./fix.patch;
      thumbsBinDir = "${placeholder "out"}/bin";
    })
  ];
```
`Not a directory: /nix/store/some-hash-fix.patch/bin/thumbs`

Reply:
@SuperSandro2000 I managed to get rid completely of this one `substitute` :)